src/pages/TrueSight/hooks/useGetCoinGeckoChartData.ts

```diff
@@ -51,32 +55,33 @@ export default function useGetCoinGeckoChartData(
             // CAKE
             url = `https://api.coingecko.com/api/v3/coins/binance-smart-chain/contract/0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82/market_chart/range?vs_currency=usd&from=1649824248&to=1649910648`
           }
-          setError(undefined)
-          setIsLoading(true)
-          setData({ prices: [], market_caps: [], total_volumes: [] })
-          const response = await fetch(url, { signal: controller.signal })
+          if (Date.now() - latestRequestingTime.current < 2000) {
+            // Too Many Request
+            throw new Error('429')
```

Comment:
should create a particular error like `ToManyRequestError`. the other components will check it to handle it's logic.

Reply:
It is my trick only, no need for a variable or constant here. If we have one more use-case like this CoinGecko, I'll refactor <3.

Reply:
Sound's good!